pandas/core/arrays/datetimes.py

```diff
@@ -1728,7 +1728,14 @@ def _generate_regular_range(cls, start, end, periods, freq):
             raise ValueError("at least 'start' or 'end' should be specified "
                              "if a 'period' is given.")
 
-        values = np.arange(b, e, stride, dtype=np.int64)
+        with np.errstate(over="raise"):
+            try:
```

Comment:
also need to be moved out to another module, this is adding way to much reader complexity here (reader means future contributor to pandas)

Reply:
OK.  I think this can also be cleaned up by making a decorator to set np.errstate once instead of repeatedly.

Reply:
any thoughts on moving offsets.generate_range to this new module?  that is the only place where it is used; thematically a much better fit here than in offsets (much of which should go up to tslibs anyways)